_posts/2017-01-21-Wonders-of-Static-Linking.md

```diff
@@ -0,0 +1,147 @@
+---
+layout: post
+title: "Wonders Of Static Linking"
+date: 2017-01-21 20:37:00
+author: lordPoseidon
+---
+
+Hi, this is Nimit Bhardwaj, a student of Computer Science, we all in the journey of computer science got the chance to learn the C Programming language, yes we all, not all although, learning C as out first programming language taught us many things, many things which are related to the working of the basics of the programming.
+
+We are taught how to compile a program, yes very easy process, most of us think its just pressing the **Compile** button or pressing the **Compile and Execute** button for those who use Dev-C/C++ or just use `gcc` who use linux. Its a good defination for a layman, but for a programmer, yes for us, its not enough. For the coders(some basic coders), like us, there are 4 steps of the compilation these are
+
+* Preprocessing
+* Compilation
+* Assembling
+* Linking
+
+The preprocessing is easy for C/C++ languages, it just replace the pre-program directive, the lines starting with `#`, like `#include <stdio.h>` with the file which has the **prototypes** of the functions which will be used or replace the Macros with their expansions like `#define POSEIDON 7`, this line will replace the occurence of POSEIDON with 7, pretty basic huh.
+
+Then is compilation, which checks the possiblities of the syntax and sementic errors if possible and then will initilize the assembling phase of the process.
+
+Then is the assembling, this will convert the language, the high level language which is C to the Assembly language, a low level language, which is hardware dependent. The C code after the preprocessing and compilation will be converted to the assembly language of the given hardware, which will be the .obj file, the object file.
+
+Then is the linking phase, the whole program even from small "Hello World" program to the big Windows like Operating systems, at some phase requires linking. So, what is linking, we usually understand or got some idea of the other processes but getting linking can be difficult for begineers, but I bet it is not difficult, what it is, is very simple concept. Now take the example, you make the "Hello World" program, 
```

Comment:
`the big Windows like Operating systems` can be replaced with something like `the
the big ones like Browsers, Document Editors etc.`